exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/errors"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+)
+
+type (
+	AzureMachineTemplate struct {
+		// VMSize is the size of the Virtual Machine to build.
+		// See https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#virtualmachinesizetypes
+		VMSize string `json:"vmSize"`
+
+		// Image is used to provide details of an image to use during Virtual Machine creation.
+		// If image details are omitted the image will default the Azure Marketplace "capi" offer,
+		// which is based on Ubuntu.
+		// +kubebuilder:validation:nullable
+		// +optional
+		Image *infrav1.Image `json:"image,omitempty"`
+
+		// OSDisk contains the operating system disk information for a Virtual Machine
+		OSDisk infrav1.OSDisk `json:"osDisk"`
+
+		// SSHPublicKey is the SSH public key string base64 encoded to add to a Virtual Machine
+		SSHPublicKey string `json:"sshPublicKey"`
+	}
+
+	// AzureMachinePoolSpec defines the desired state of AzureMachinePool
+	AzureMachinePoolSpec struct {
+		// Location is the Azure region location e.g. westus2
+		Location string `json:"location"`
+
+		// Template contains the details used to build a replica virtual machine within the Machine Pool
+		Template AzureMachineTemplate `json:"template"`
+
+		// AdditionalTags is an optional set of tags to add to an instance, in addition to the ones added by default by the
+		// Azure provider. If both the AzureCluster and the AzureMachine specify the same tag name with different values, the
+		// AzureMachine's value takes precedence.
+		// +optional
+		AdditionalTags infrav1.Tags `json:"additionalTags,omitempty"`
```

Comment:
Is there a particular reason to qualify tags as additional?

Reply:
The rational we used for this naming in cluster-api-provider-aws was that tags specified in our field do not replace the provider-defined tags, but are "in addition to" them.

Reply:
same here, these tags are merged with the default values, they define the set of tags to be applied but rather tags to be added to the default ones.